File: Anushka-Pachauri/SlidingWindow/FindAllAnagramsInAString.java
class FindAllAnagramsInAString. {
    public List<Integer> findAnagrams(String s, String p) {
        char[] s1=s.toCharArray();
        char[] p1=p.toCharArray();
        int[] sFreq=new int[26];
        int[] pFreq=new int[26];
        List<Integer> answer=new ArrayList<>();

        if(s1.length<p1.length){
            return new ArrayList<>();
        }

        for(int i=0;i<p1.length;i++){
            pFreq[p1[i]-'a']++;
            sFreq[s1[i]-'a']++;
        }
        if(Arrays.equals(sFreq,pFreq)){
            answer.add(0);
        }

        int start=0;
        int end=p1.length;
        while(end<s1.length){
            sFreq[s1[start]-'a']--;
            sFreq[s1[end]-'a']++;
            if(Arrays.equals(sFreq,pFreq)){
                answer.add(start+1);
            }
            start++;
            end++;
        }
        return answer;
    }
}

File: NikhilKumarGupta/Slidingwindow/Find All Anagrams in a string.cpp
class Solution {
    public:
        vector<int> findAnagrams(string s, string p) {
            vector<int>window(26,0),pfreq(26,0);
            int start=0;
            vector<int>ans;
            if(p.size()>s.size())return ans;
            
    
            for(int i=0;i<p.size();i++){
                window[s[i]-'a']++;
                pfreq[p[i]-'a']++;
    
            }
            if(window==pfreq){
                ans.push_back(start);
            }
            for(int i=p.size();i<s.size();i++){
                window[s[i]-'a']++;
                window[s[i-p.size()]-'a']--;
    
                 if(window==pfreq){
                ans.push_back(i-p.size()+1);
            }
                
            }
            return ans;
    
    
            
            
        }
    };